### DISCUSSION TITLE
Kubernetes packages

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [dotnet/yarp#1523] [Q&A] [answered]
Are the Kubernetes-related packages supposed to be available on nuget? The docs list various "packages" and their dependencies, but when I do a nuget search for Yarp.Kubernetes.Ingress, Yarp.Kubernetes.Controller or Microsoft.Kubernetes.Core. I don't get any results.

We'd like to hopefully start using YARP integrated with Kubernetes. I imagine that I'd have to clone the Controller project as there's some tweaks we'd wanna make that are specific to our use case (populating various settings based on annotations/flowing annotations as custom metadata), but I was really hoping not to have to copy all of the Operator framework and the k8s core library as well. 

Am I missing something?

**Comment by MihaZupan** [comment, marked as answer]
We are working with the community to improve the Kubernetes implementation, but we have not published any packages for it yet.
All the samples/dockerfiles are buildable locally if you are interested in trying them out and providing any feedback.

**Reply by specialforest** [reply]
Perhaps that's more "dotnet/arcade" question, which I'm not familiar with... Is there a way to build them locally?

That's what I have been doing so far.
Removing the following from all Kubernetes related projects:
```csproj
<IsPackable>$('System.TeamProject') != 'internal'</IsPackable>
```
and adding
```csproj
<OutputType>Library</OutputType>
```

As well as adding the following to src\OperatorFramework\Directory.Build.props:
```csproj
<PropertyGroup>
  <IsShipping>true</IsShipping>
  <IsPackable>true</IsPackable>
</PropertyGroup>
```

And finally passing the following additional arguments to pack.cmd:
```
-c Release "/p:VersionSuffix=mysuffix"
```

**Reply by Tratcher** [reply]
Alternatively you could check out this repo as a submodule and use direct project references.
